src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,158 @@
+import React, {
+  useCallback, useEffect, useRef, useState,
+} from 'react';
+import classNames from 'classnames';
+
+import { Todo } from '../../types/Todo';
+import { useTodo } from '../../hooks/useTodo';
+import { ErrorMessage } from '../../types/ErrorMessage';
+import { deleteTodo } from '../../api/todos';
+
+type Props = {
+  todo: Todo;
+  loading?: boolean;
+};
+
+export const TodoItem: React.FC<Props> = ({ todo, loading }) => {
+  const {
+    todos,
+    setTodos,
+    setIsChecked,
+    setErrorMessage,
+    isProcessing,
+    setIsProcessing,
+  } = useTodo();
+  const [title, setTitle] = useState(todo.title);
+  const [isEditing, setIsEditing] = useState(false);
+  const [focus, setFocus] = useState(false);
```

Comment:
Rename all boolean variables according [tips](https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig)
```suggestion
  const [isFocus, setIsFocus] = useState(false);
```